src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -253,6 +253,10 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
 }
 
 /** Base trait for strict collections that can be built using a builder.
+  *
+  * All factories provided by such a collection must be of the `*WithBuilder` variant
+  * even though this is not formally enforced through types.
+  *
```

Comment:
We can relax it from *must* to *should* though. Some strict collections may not be able to provide a more efficient builder anyway so it would be OK to fall back to a an ArrayBuffer. Implementing a `WithBuilder` factory would be an optimization, just like overriding a default implementation of some collection operation.